core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -951,6 +951,25 @@ wasi_path_remove_directory(wasm_exec_env_t exec_env, wasi_fd_t fd,
     return wasmtime_ssp_path_remove_directory(curfds, fd, path, path_len);
 }
 
+static int
+get_timeout_for_poll_oneoff(const wasi_subscription_t *in)
+{
+    int timeout = 0;
+    const __wasi_subscription_t *clock_subscription = NULL;
+    if (clock_subscription == NULL
+        && (in->u.u.clock.flags & __WASI_SUBSCRIPTION_CLOCK_ABSTIME) == 0) {
+        clock_subscription = in;
+    }
+    if (clock_subscription != NULL) {
+        __wasi_timestamp_t ts = clock_subscription->u.u.clock.timeout / 1e9;
```

Comment:
should this be 1e9 or 1e6 as written [here](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c#L2693)?

Reply:
the timeout is in `nanoseconds`, so if you want to convert it to seconds, it should be `1e9`.

Reply:
right, was unsure why it was 1e6 above because of the same reason